code/modules/food_and_drinks/food/foods/misc_food.dm

```diff
@@ -114,9 +114,41 @@
 	filling_color = "#DEDEAB"
 	list_reagents = list("nutriment" = 4)
 	tastes = list("meat" = 2, "dough" = 2, "laziness" = 1)
+	var/warm = TRUE
+	var/cool_timer
+	var/heated_at
+	var/subtracted_time = 0
+
+/obj/item/reagent_containers/food/snacks/warmdonkpocket/Initialize(mapload)
+	. = ..()
+	donk_heat()
+
+/obj/item/reagent_containers/food/snacks/warmdonkpocket/proc/donk_heat()
+	heated_at = world.time
+	cool_timer = addtimer(CALLBACK(src, PROC_REF(donk_cool)), 7 MINUTES - subtracted_time, TIMER_UNIQUE|TIMER_STOPPABLE)
+
+/obj/item/reagent_containers/food/snacks/warmdonkpocket/proc/donk_cool()
+	name = "cold Donk-pocket"
+	desc = "The food of choice for the seasoned traitor. This one is cold."
+	warm = FALSE
+
+/obj/item/reagent_containers/food/snacks/warmdonkpocket/on_enter_storage(obj/item/storage/S)
+	. = ..()
+	if(istype(S, /obj/item/storage/box/donkpockets) && cool_timer)
+		deltimer(cool_timer)
+		subtracted_time += (world.time - heated_at)
+		cool_timer = null
```

Comment:
Guard clause please

Reply:
How would I apply guard clause here? I thought that was for chains of if-else if which use early returns

Reply:
It should early return if it doesn't have a timer instead of always doing a type check.